src/brokers/utils/add-broker.ts

```diff
@@ -18,8 +22,40 @@ export const BrokerCreationFormState = createContext<FormState>({});
 export const BrokerCreationFormDispatch =
   createContext<React.Dispatch<ArtemisReducerActions>>(null);
 
+const artemisCRStateMap = new Map<string, FormState>();
+
+const initialCr = (namespace: string, name: string): ArtemisCR => {
+  return {
+    apiVersion: 'broker.amq.io/v1beta1',
+    kind: 'ActiveMQArtemis',
+    metadata: {
+      name: name,
+      namespace: namespace,
+    },
+    spec: {},
+  };
+};
+
+export const getArtemisCRState = (name: string, ns: string): FormState => {
+  const key = name + ns;
+  let formState = artemisCRStateMap.get(key);
+  if (!formState) {
+    formState = {};
+    formState.shouldShowYAMLMessage = true;
+    formState.editorType = EditorType.BROKER;
+    formState.editorState = {
+      stringValues: new Map<string, string>(),
+      booleanValues: new Map<string, boolean>(),
+    };
```

Comment:
I understand that this is used so that the editor remembers what property is being worked on, so that when the user goes back and forth between the configurator and the yaml editor, the last open view stays open. I think this is a great and important addition.

However, I'm not sure if we should work with open fields in the reducer. As it means that we're extracting the logic out of the editor. See Later on in the code the state is changed as followed:

```typescript
  const onSelectBrokerConfigItem = (
    selectedItem: any,
    selectedItemProps: any,
  ) => {
    console.log('new selection selected', selectedItem);
    dispatch({
      operation: ArtemisReducerOperations.updateEditorState,
      payload: {
        stateType: StateType.String,
        itemId: configId,
        state: selectedItemProps.itemId,
      },
    });
  };
```
The operation `updateEditorState` being generic, we can't know what is being changed on the state unless performing a complete examination of the code.

Instead I would prefer to have:

```typescript
  const onSelectBrokerConfigItem = (
    selectedItem: any,
    selectedItemProps: any,
  ) => {
    dispatch({
      operation: ArtemisReducerOperations.setEditorActiveProperty,
      payload: selectedItemProps.itemId,
      },
    });
  };
```

In this case, we the operation is named according to the action it does, and the payload can be documented accordingly to: "set the active property the user is working on."